### DISCUSSION TITLE
Option to propagate setting/permissions changes to rooms

### DISCUSSION THREAD

**Discussion by SethFalco** [element-hq/element-meta#582] [Ideas]
#### What would you like to do?

When changing settings in a Space, especially permissions, I'd like to have an easy button that allows me to propagate those changes to all rooms in the space that I also have permission to manage that setting for.

#### Why would you like to do it?

Sometimes when I change settings, it's something I want to change across the entire space including nested chat rooms. Mostly for permissions.

This would also be handy for assigning a role to a user, to propagate that role to all rooms and spaces under the space. (At least for the Moderator and Admin, but in general it could be nice for all nested rooms/spaces where a role with that name exists.)

#### How would you like to achieve it?

I'm not a UI/UX designer, but maybe a viable approach based on the current desktop UI is to add a **Save** button to the bottom, with a select-menu/button similar to what GitHub merge requests have. By default, it's just **Save**, but can have an alternate selection to save and propagate changes down.

### Have you considered any alternatives?

There could be a permissions specific UI, once again based on the current desktop design, a possible alternative could be that on change, a button appears to the right of the permission to propagate the change down.

### Additional context

This would hopefully make managing larger rooms less tedious.

Ideally, this would work similarly to how when you want to kick a user from a space, there is a modal to **Remove them from everything I'm able to**.

**Comment by Pilzinsel64** [comment]
Are there any working alternatives for the time beeing, via an bot for example?
